src/api/MessageEvents.ts

```diff
@@ -41,16 +41,16 @@ export interface MessageExtra {
     stickerIds?: string[];
 }
 
-export type SendListener = (channelId: string, messageObj: MessageObject, extra: MessageExtra) => void | { cancel: boolean; };
-export type EditListener = (channelId: string, messageId: string, messageObj: MessageObject) => void;
+export type SendListener = (channelId: string, messageObj: MessageObject, extra: MessageExtra) => Promise<void | { cancel: boolean; }>;
```

Comment:
```suggestion
export type SendListener = (channelId: string, messageObj: MessageObject, extra: MessageExtra) => Promiseable<void | { cancel: boolean; }>;
```

Just make the promise aspect optional instead of forcing plugins to make their listeners async